include/boost/url/impl/url_base.ipp

```diff
@@ -1337,6 +1337,108 @@ end() const noexcept ->
     return iterator(v_, true);
 }
 
+url_base::segments_type::iterator
```

Comment:
Use `auto` and trailing return type here so we don't have to utter a long type name.